services/notifier/repo.go

```diff
@@ -303,6 +306,19 @@ func (n *repo) deleteByBatchID(
 	if err != nil {
 		return fmt.Errorf("deleting by batchID: %w", err)
 	}
+
+	var sizeEstimate int64
+	if err := n.db.QueryRowContext(
+		ctx,
+		fmt.Sprintf(`SELECT pg_table_size(oid) from pg_class where relname='%s';`, notifierTableName),
+	).Scan(&sizeEstimate); err != nil {
+		return fmt.Errorf("size estimate for notifierTable failed with: %w", err)
+	}
+	if sizeEstimate > config.GetInt64("Notifier.vacuumThresholdBytes", 5*bytesize.GB) {
+		if _, err := n.db.ExecContext(ctx, fmt.Sprintf(`vacuum full analyze %q;`, notifierTableName)); err != nil {
```

Comment:
Did the full vacuuming helped here?

Reply:
Yes it did. When the size got high the DB queries got slower affecting the throughput, so added the vacuum command to automate it

Reply:
Should we do it from a master instead of a slave once all the track batch is completed?

Reply:
Let's merge this and take that up if this causes any issues?